bin/oneoff/purge_accounts/purge_users

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env ruby
+
+# This script will purge accounts given a CSV file with the column of 'user_id'.
+# You must supply the CSV file as the first argument and it will do a "dry run"
+# and report found and not found ids. Supplying "for-real" as the second
+# argument will perform the purge.
+
+# Refer to doc for information about how to request and fulfill such
+# requests and use this script:
+# https://docs.google.com/document/d/1-Gk66udFdzqvYyT6J-1d5HPSbkVCIQWzxMLyVKF3fuk/edit
+
+if ARGV.length.empty? || ARGV.length > 2
+  puts 'Usage: ./bin/oneoff/purge_accounts/purge_users ./bin/oneoff/purge_accounts/yyyy-mm-dd-users.csv [commit]'
+  puts 'The CSV needs a column with "user_id" with User ids.'
+  puts 'Will do a "dry run" until you specify "for-real" for the "commit" field.'
+  exit 1
+end
+
+require 'csv'
+
+ids = []
+begin
+  csv_file = ARGV[0]
+  CSV.foreach(csv_file, headers: true, col_sep: ",") do |row|
+    ids << row["user_id"] if row["user_id"]
+  end
+rescue
+  puts "Problem reading given CSV file."
+end
+
+if ids.empty?
+  puts "Could not find any user ids. The CSV needs a column with 'user_id' with User ids."
+  exit 1
+end
+
+puts "Found #{ids.count} ids to purge."
+
+require_relative '../../../dashboard/config/environment'
+require_relative '../../../dashboard/lib/account_purger'
```

Comment:
Could be my lack of ruby knowledge, but is there a reason why we wouldn't put `require`s at the top of the file? In other languages it's usually best practice to declare them at the top to help with readability and maintainability.

Same for line 19 (`require 'csv'`)

Reply:
Oh, because they take a long time. If you put them at the top of the file, it would spin for ten minutes as it loaded our entire app and then tell you that are argument is wrong.

Reply:
I will move the 'csv' require to the top.